Packs/OktaASA/Integrations/OktaASA/OktaASA.py

```diff
@@ -0,0 +1,359 @@
+
+        Returns:
+            List[Dict]: events
+            str: id for last run
+        """
+        results: List[Dict] = []
+        descending = False
+        returned_offset = offset
+        returned_timestamp = None
+        # We are limited to 1000 results per request, count > 1000 does not work.
+        count = min(limit, 1000) if limit else 1000
+        while limit and len(results) < limit:
+            descending = bool(not returned_offset)
+            events = self.execute_audit_events_request(
+                offset=returned_offset, count=count, descending=descending, prev=None
+            )
+            if not events:
+                break
+            event_offset = events[0] if descending else events[len(events) - 1]
+            returned_offset = event_offset.get("id")
+            results.extend(events)
+            count = min(limit - len(results), 1000)
+        if results:
+            list_last_index = len(results) - 1
+            demisto.debug(f"{INTEGRATION_NAME}: will return {len(results)} events")
+            returned_timestamp = results[list_last_index].get("timestamp")
+
+        return results, returned_offset, returned_timestamp
+
+
+'''HELPER FUNCTIONS'''
+
+
+def is_token_expired(expires_date: str) -> bool:
+    """Checks if token is expired.
+
+        Args:
+            self (OktaASAClient): Okta ASA Client.
+            expires_date (str): The expiration date.
+        Returns:
+            bool: is the token expired.
+    """
+    current_utc_time = datetime.now(pytz.utc)
+    expires_datetime_date = dateparser.parse(expires_date, settings={'TIMEZONE': 'UTC'}) or current_utc_time
+    expires_datetime_date = expires_datetime_date - timedelta(hours=0, minutes=3)
+    expires_datetime: datetime = arg_to_datetime(expires_date) or current_utc_time
+    return current_utc_time > expires_datetime
+
+
+def add_time_to_events(events: List[Dict] | None):
+    """
+    Adds the _time key to the events.
+    Args:
+        events: List[Dict] - list of events to add the _time key to.
+    Returns:
+        list: The events with the _time key.
+    """
+    if events:
+        for event in events:
+            create_time = arg_to_datetime(arg=event.get('timestamp'))
+            event['_time'] = create_time.strftime(DATE_FORMAT) if create_time else None
+
+
+'''COMMAND FUNCTIONS'''
+
+
+def test_module(client: OktaASAClient, team_name: str) -> str:
+    """
+    Tests API connectivity and authentication
+    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
+    successful.
+    Raises exceptions if something goes wrong.
+
+    Args:
+        client (OktaASAClient): OktaASAClient client to use.
+        team_name (str): the name of the team.
+
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+    try:
+        fetch_events_command(
```

Comment:
No need to run the fetch events command, it uses the last run... we can call the get events, or directly to the client method which can get events from the API